ERCS/erc-7589.md

```diff
@@ -0,0 +1,504 @@
+  * The commitment is not found (no tokens were committed).
+  * The `msg.sender` is not and was not approved by the grantor.
+  * The commitment has at least one non-revocable role that didn't expire.
+* The `setRoleApprovalForAll` function MAY be implemented as public or external.
+* The `grantorOf` function MAY be implemented as pure or view and MUST return the owner of the committed tokens.
+* The `tokenAddressOf` function MAY be implemented as pure or view and MUST return the address of the committed tokens.
+* The `tokenIdOf` function MAY be implemented as pure or view and MUST return the identifier of the committed tokens.
+* The `tokenAmountOf` function MAY be implemented as pure or view and MUST return the token amount committed.
+* The `roleData` function MAY be implemented as pure or view and MUST return the custom data of the role assignment.
+* The `roleExpirationDate` function MAY be implemented as pure or view and MUST return the expiration date of the role 
+  assignment.
+* The `isRoleRevocable` function MAY be implemented as pure or view and MUST return whether the grantor can end the role
+  assignment before its expiration date.
+* The `isRoleApprovedForAll` function MAY be implemented as pure or view and MUST return whether the `_operator` is
+  allowed to grant and revoke roles on behalf of the `_grantor`.
+
+> Please note that "approval" refers to allowing users to commit tokens and grant/revoke roles on one's behalf. An
+  approved user either received the role approval or is the target user. Role approvals are not to be confused with
+  [ERC-1155](./eip-1155.md) approvals. More information can be found in the [Role Approvals](#role-approvals) section.
+
+## Rationale
+
+The concept of "token commitments" as an abstraction serves as a powerful tool for users looking to delegate the control
+of their SFTs. A token commitment represents either a frozen balance or tokens deposited into a contract, offering a
+standardized and secure way for SFT owners to delegate the use of their assets. Through [ERC-7589](./eip-7589.md), users
+gain a versatile mechanism to abstract the complexities of secure delegation, enhancing the utility and interoperability
+of semi-fungible tokens.
+
+[ERC-7589](./eip-7589.md) IS NOT an extension of [ERC-1155](./eip-1155.md). The main reason behind this decision is to
+keep the standard agnostic of any implementation. This approach enables the standard to be implemented externally or on
+the same contract as the SFT and allows dApps to use roles with immutable SFTs.
+
+### Role Approvals
+
+Like [ERC-1155](./eip-1155.md), [ERC-7589](./eip-7589.md) allows users to approve operators to grant and revoke roles on
+their behalf. This feature is crucial for interoperability, as it enables third-party applications to manage user roles
+without custody-level approvals. Role approvals are part of the core interface, and compliant contracts MUST implement
```

Comment:
The "MUST" (all caps) here is creating a requirement, and so should be in the Specification section.

If you are _explaining_ the requirement already introduced in the Specification section, don't use ALL CAPS.